qemu/tests/win_video_play.py

```diff
@@ -0,0 +1,81 @@
+import time
+
+from avocado.core import exceptions
+from virttest import error_context
+from virttest import utils_misc
+
+
+@error_context.context_aware
+def run(test, params, env):
+
+    def check_wmplayer_installed(session):
+        """
+        Check if wmplayer is installed
+        """
+        error_context.context("Check if wmplayer is installed", logging.info)
+        install_path = params.get("wmplayer_path")
+        check_cmd = 'dir "%s"|findstr /I wmplayer'
+        check_cmd = params.get("wmplayer_check_cmd", check_cmd) % install_path
+        if session.cmd_status(check_cmd) == 0:
+            return install_path
+
+    def check_kmplayer_installed(session):
+        """
+        Check if kmplayer is installed
+        """
+        error_context.context("Check if kmplayer is installed", logging.info)
+        install_path = params.get("kmplayer_path")
+        check_cmd = 'dir "%s"|findstr /I kmplayer'
+        check_cmd = params.get("kmplayer_check_cmd", check_cmd) % install_path
+        if session.cmd_status(check_cmd) != 0:
+            kmplayer_install(session)
+        return install_path
+
+    def kmplayer_install(session):
+        """
+        Install kmplayer
+        """
+        error_context.context("Install kmplayer ...", logging.info)
+        guest_name = params["guest_name"]
+        alias_map = params["guest_alias"]
+        guest_list = dict([x.split(":") for x in alias_map.split(",")])
+        guest_name = guest_list[guest_name]
+
+        install_cmd = params["kmplayer_install_cmd"] % guest_name
+        install_cmd = utils_misc.set_winutils_letter(session, install_cmd)
+        s, o = session.cmd_status_output(install_cmd)
+        if s != 0:
+            raise exceptions.TestError("Failed to install kmplayer %s" % o)
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = float(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
+
+    wmplayer = check_wmplayer_installed(session)
+    if wmplayer:
+        video_player = wmplayer
+        if params.get("wmplayer_reg_cmd"):
+            logging.info("Update regedit")
+            session.cmd(params.get("wmplayer_reg_cmd"))
+    else:
+        kmplayer = check_kmplayer_installed(session)
+        video_player = kmplayer
+        if params.get("kmplayer_reg_cmd"):
+            error_context.context("Update register", logging.info)
+            session.cmd(params.get("kmplayer_reg_cmd"))
+            time.sleep(5)
+
+    video_url = params["video_url"]
+    play_video_cmd = params["play_video_cmd"] % (video_player, video_url)
+    error_context.context("Play video", logging.info)
+    try:
+        session.cmd(play_video_cmd, timeout=300)
+        time.sleep(params.get("time_for_video", 240))
+    except Exception, details:
+        raise exceptions.TestFail(details)
+    finally:
+        error_context.context("Stop video", logging.info)
+        session.cmd('del /f /s "%s"' % video_player)
```

Comment:
1. I tried to raise error in "try".
1.1 I have updated timeout to 1 in "session.cmd(play_video_cmd, timeout=300)", test pass.
1.2 I have updated the video_url to a non-existing one, test pass.
2. I tried to raise error in "finally", just updated "del" to "el" which is a wrong cmd, test failed, I think such errors should be ignored as this is not the key point.
700 21:09:18 ERROR| Traceback (most recent call last):
701 21:09:18 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-44.0-py2.7.egg/avocado_vt/test.py", line 196, in runTest
702 21:09:18 ERROR|     self._runTest()
703 21:09:18 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-44.0-py2.7.egg/avocado_vt/test.py", line 337, in _runTest
704 21:09:18 ERROR|     run_func(self, params, env)
705 21:09:18 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-44.0-py2.7.egg/virttest/error_context.py", line 135, in new_fn
706 21:09:18 ERROR|     return fn(*args, **kwargs)
707 21:09:18 ERROR|   File "/usr/share/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/win_video_play.py", l    ine 80, in run
708 21:09:18 ERROR|     session.cmd('el /f /s "%s"' % video_player)
709 21:09:18 ERROR|   File "/usr/lib/python2.7/site-packages/aexpect/client.py", line 1211, in cmd
710 21:09:18 ERROR|     raise ShellCmdError(cmd, s, o)
711 21:09:18 ERROR| ShellCmdError: Shell command failed: 'el /f /s "%ProgramFiles%\\Windows Media Player\\wmplayer.exe"'    (status: 9009,    outp    ut: "'el' is not recognized as an internal or external command,\noperable program or batch file.\n\n")
712 21:09:18 ERROR|
713 21:09:18 ERROR| Traceback (most recent call last):
714 
715 21:09:18 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-44.0-py2.7.egg/avocado_vt/test.py", line 434, in _run_avocado
716     raise test_exception
717 
718 21:09:18 ERROR| ShellCmdError: Shell command failed: 'el /f /s "%ProgramFiles%\\Windows Media Player\\wmplayer.exe"'    (status: 9009,    outp    ut: "'el' is not recognized as an internal or external command,\noperable program or batch file.\n\n")

Anyway, as I said, this is an enhancement, it up to you.